manuskript/ui/editors/editorWidget.py

```diff
@@ -313,17 +312,26 @@ def setCurrentModelIndex(self, index=None):
         if self._model:
             self.setView()
 
-    def updateIndexFromID(self):
+    def updateIndexFromID(self, fallback=None, ignore=None):
         """
         Index might have changed (through drag an drop), so we keep current
         item's ID and update index. Item might have been deleted too.
+
+        It will ignore the passed model item to avoid ambiguity during times
+        of inconsistent state.
         """
-        idx = self._model.getIndexByID(self.currentID)
 
-        # If we have an ID but the ID does not exist, it has been deleted
+        idx = self._model.getIndexByID(self.currentID, ignore=ignore)
+
+        # If we have an ID but the ID does not exist, it has been deleted.
         if self.currentID and idx == QModelIndex():
-            # Item has been deleted, we open the parent instead
-            self.setCurrentModelIndex(self.currentIndex.parent())
+            # If we are given a fallback item to display, do so.
+            if fallback:
+                self.setCurrentModelIndex(fallback)
+            else:
+                #self.mw.mainEditor.tabSplitter.tab.removeTab(self.mw.mainEditor.tab.indexOf(self))
+                self.setCurrentModelIndex(None) # TODO: change this to close the tab.
```

Comment:
You'll be removing this since it's unused code, but this would cause a crash too since the first thing setCurrentModelIndex does is "index.isValid()" so passing None to it would crash.

Reply:
I left it in because I do not have a 'proper' fix for that bit of code without digging all the way into the editor code. This function can't start trying to dig into the existing item and obtain a parent because we'll end up where we were, and removing it entirely will leave that tab in an inconsistent (and likely crashy) state.

The code as it is currently implemented cannot trigger the else, which is why I kind of left it like that. I can make the else `raise NotImplementedError("implement tab closing")` if you'd prefer that approach? It is probably neater.

(The neatest is to just implement it right now, but scope creep...)

Reply:
Yeah, I think if you want to leave it, put it in a comment and raise a NotImplementedError, it's better than leaving it like that because if the else ends up being used a year from now, and it causes a crash, nobody will understand why that happens.